crates/parcel_napi_helpers/src/transferable.rs

```diff
@@ -0,0 +1,81 @@
+use std::any::Any;
+use std::collections::HashMap;
+use std::marker::PhantomData;
+use std::sync::atomic::AtomicI32;
+use std::sync::atomic::Ordering;
+use std::sync::Mutex;
+
+use napi::bindgen_prelude::FromNapiValue;
+use napi::bindgen_prelude::ToNapiValue;
+use napi::Env;
+use napi::JsNumber;
+use napi::NapiRaw;
+use once_cell::sync::Lazy;
+
+static COUNTER: AtomicI32 = AtomicI32::new(0);
+static VALUES: Lazy<Mutex<HashMap<i32, Box<dyn Any + Send + Sync>>>> =
+  Lazy::new(|| Default::default());
+
+/// Creates an external reference to a Rust value and
+/// makes it transferable across Nodejs workers
+///
+/// This is to get around the limitations of what can be transferred
+/// between workers in Nodejs
+///
+/// https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Transferable_objects
+pub struct JsTransferable<T> {
+  id: i32,
+  _value: PhantomData<T>,
+}
+
+impl<T: Send + Sync + 'static> JsTransferable<T> {
+  /// Put a Rust value into a Transferable container to allow
+  /// sending values to Nodejs workers via postMessage or workerData
+  pub fn new(value: T) -> Self {
+    let id = COUNTER.fetch_add(1, Ordering::Relaxed);
+
+    VALUES.lock().unwrap().insert(id.clone(), Box::new(value));
+    Self {
+      id,
+      _value: Default::default(),
+    }
+  }
+
+  /// Take value out of Transferable. It can no longer be accessed
```

Comment:
The first line of any doc comment should be one sentence

```suggestion
  /// Take the value out of Transferable, so it can no longer be accessed
```